### DISCUSSION TITLE
Lost audio session recovery

### DISCUSSION THREAD

**Discussion by Bert-Proesmans** [File-New-Project/EarTrumpet#1180] [Ideas]
What?

Can a facility be implemented to recover "lost" audio sessions?  
I imagine the following; 
* a menu item, or shortcut, or interface button next to device, to reattach orphaned audio sessions to the default/"contextually targetted" audio output device.
* an option to enable/disable automatic switching of sessions on device status change

Why?

When my bluetooth earbuds are worn I manually switch applications from my default output device to the bluetooth device.
When I put the earbuds in the charging case, mind that they have no specific off button, windows still sees sees them as "linked". What "linked" exactly means seems to be more akin to "known" than "actively communicating".

![afbeelding](https://user-images.githubusercontent.com/2027877/174903369-059978ec-545d-416d-890a-afa5ab421ea5.png)
NOTE; Bluetooth device says "linked"

But EarTrumpet sees no audio devices, and windows audio manager says the bluetooth device is not connected. The audio sessions do stick to the device though!

![afbeelding](https://user-images.githubusercontent.com/2027877/174903470-3c0834fe-9649-49a9-aaf1-43b2440eade3.png)
NOTE; Output device status says "Lost connection"

![afbeelding](https://user-images.githubusercontent.com/2027877/174904620-63e0068e-8bd5-4a27-aaec-13f2a27ed784.png)
NOTE; There is no Firefox audio session, it's still attached to the bluetooth audio device which is invisible

My issue is that the audio sessions are not automatically switched to the default output device. There does not seem to be an easy manual way to remove and re-attach those lost audio sessions.

Workarounds exist but are clunky;
* Delete the bluetooth device, followed by pairing again next use
* Wonkily reconnect the earbuds (windows and the earbuds only properly connect about 1 in 4 times), and manually move the sessions back to default audio output

Under normal circumstances the Bluetooth device is communicating and acts as an audio output device sink. And the attached audio sessions are visible.

![afbeelding](https://user-images.githubusercontent.com/2027877/174904356-405c0371-da0c-4185-a629-6fc79e199ef1.png)
NOTE; Bluetooth device is parent/sink of two audio output devices

![afbeelding](https://user-images.githubusercontent.com/2027877/174904545-0eca2169-f201-460d-8777-70171b6dd713.png)
NOTE; The audio output device linked to the Bluetooth device is visible, that device has the expected Firefox audio session attached

I'm aware this issue is not related to Eartrumpet, and Eartrumpet might not have a solution ready. But I figured to ask anyway since the app in itself is very robust and well built!
The earbuds themselves might be the underlying cause, I have no experience with other Bluetooth devices to compare.

Specs?

* Windows 10 21H2
* Asus BT-500 USB adapter
* Samsung Galaxy Buds 2
* Eartrumpet, latest version

Feel free to chime in or ask additional questions!

**Comment by riverar** [comment]
Hi @Bert-Proesmans, thanks for the detailed write up. Can you also file a Feedback Hub issue under **Devices and Drivers > Audio and sound**, and share the link to the issue here?

Microsoft recently made Bluetooth audio changes and I wonder if this is a regression/bug. I don't think Windows should be sending audio to inactive Bluetooth audio devices. Do put a link to this discussion in the Feedback Hub issue too.

**Reply by Bert-Proesmans** [reply]
Link to my Windows Feedback Hub issue, see https://aka.ms/AAhc5np

**Reply by Bert-Proesmans** [reply]
Hi @riverar, nothing came from the feedback hub issue so far.
The problem still exists today (Windows 10 21H2 19044.2130); Windows does not reattach audio sessions to the default output sink after disconnecting my Bluetooth wireless earbuds.

Workarounds still work;
* either unpair/remove the Bluetooth device, windows automatically moves all sessions to the default audio output sink
* either connect Bluetooth device, and manually change audio session to other audio output sink 

Guess I'm unlucky. I'm gonna leave it at this report.
